carbon-design-system/carbon-addons-iot-react
# fix: PageTitleBar description

Closes #3785 

**Summary**
Fix of incorrect behavior of moving the `PageTitleBar` description to a tooltip when content was provided. It is now displayed below the title and only moved to the tooltip if the `collapsed` prop is passed.
I've also added a feature requested by @JordanWSmith15 where the condensed variation of the title bar (with breadcrumbs) now contains the tooltip as well so the description can be easily accessed.

**Change List (commits, features, bugs, etc)**
- PageTitleBar logic fix to show description when content is provided and only move it to Tooltip if the condensed prop is true
- Tooltip with description added to condensed variation of page title bar to be easily accessed when header is "Dynamic" or "Condensed"
- Unit tests added and updated
- Snapshots updated
- PageTitleBar e2e tests fixed after changes

**Acceptance Test (how to verify the PR)**
- Check PageTitleBar stories and add a description to some of them. Unless `condensed` is set to true the description should be displayed below the title. For the DYNAMIC variation, it should be moved to the tooltip next to the current breadcrumb item when scrolling down. For the CONDENSED variation, it should only be in the tooltip next to the current breadcrumb item.

**Regression Test (how to make sure this PR doesn't break old functionality)**
- Check if description below title is not rendered when it shouldn't like when scrolling down with DYNAMIC headerMode or when using CONDENSED headerMode



**Things to look for during review**
